.github/workflows/QC.yml

```diff
@@ -0,0 +1,12 @@
+name: Quality Check
+on: 
+  workflow_call: 
+jobs: 
+  qualitycheck: 
+    name: Quality Check
+    runs-on: ubuntu-20.04
+    steps:
+    - name: Checkout source
+      uses: actions/checkout@v3
+    - name: Execute Rustfmt check
+      run: cd src && cargo fmt --all
```

Comment:
`cargo fmt --all` is used for formatting the source code, which is not what we want here. What we want is `cargo fmt --check` to check if the source code is in a valid style.